onnxruntime/core/session/inference_session.h

```diff
@@ -46,6 +46,12 @@ namespace logging {
 class LoggingManager;
 }
 
+struct FreeDimensionOverride
+{
+  std::string dimension_denotation;
```

Comment:
Since this is going to be exposed in the C API as well, can you use std::basic_string<ORTCHAR_T> like other strings in SessionOptions? Examples are optimized_model_filepath and profile_file_prefix.

Reply:
This string represents an ONNX dimension denotation (e.g [onnx::DATA_BATCH](https://github.com/onnx/onnx/blob/master/onnx/common/constants.h#L17)). Shouldn't that always be a regular char string? Either way is fine, just double-checking. :)

Reply:
I was thinking more in terms of dim_param. If these are going to be just denotations (based on the previous discussion), I guess this is ok.